src/NewProject/source_grid.cpp

```diff
@@ -364,6 +404,18 @@ void sourceGrid::onItemChanged(QStandardItem *item) {
     m_lisFileData[itemIndex.row()].m_importLibraries = item->text();
 }
 
+void sourceGrid::languageHasChanged() {
+  QComboBox *combo = qobject_cast<QComboBox *>(sender());
```

Comment:
Currently not a problem because we are connecting to a QComboBox, but to stay safe, probably best to check that the qobject_cast succeeded (combo isn't null) incase someone accidentally connects a non-qcombobox widget to this signal later on.

Reply:
good catch! thanks. Fixed